### PR TITLE
fix gateio fetchOHLCV

### PR DESCRIPTION
**Pull request by frosty00**

### CODE REVIEW

File: js/gateio.js [hunk 1]
```diff
@@ -770,7 +771,7 @@ module.exports = class gateio extends Exchange {
     }
 
     prepareRequest (market) {
-        if (market['contract']) {
+        if (market['derivative']) {
```
Comment (samgermain) [inline js/gateio.js:774]: Why is this changed? `market['contract']` is used everywhere else

Comment (frosty00) [reply js/gateio.js:774]: we've changed `contract` to mean the contractSize is not one, aka you have to be careful when making orders and derivative means futures aka swap or future

File: js/gateio.js [hunk 2]
```diff
@@ -649,13 +650,13 @@ module.exports = class gateio extends Exchange {
                         'swap': swap,
                         'option': option,
                         'derivative': true,
-                        'contract': true,
+                        'contract': (contractSize !== '1'),
```
Comment (samgermain) [inline js/gateio.js:653]: Whats the logic behind this?

File: js/gateio.js (hunk 1 above)
Comment (samgermain) [reply js/gateio.js:774]: > we've changed contract to mean the contractSize is not one,

Why is that what it means?

Comment (kroitor) [reply js/gateio.js:774]: @frosty00 i think it's a different meaning right now. Specifically, this:

> We could have derivative: true for all contracts and contract: true only for those where you trade contracts (amount precision in contracts).

File: js/gateio.js (hunk 2 above)
Comment (kroitor) [reply js/gateio.js:653]: @frosty00 I would agree with Sam – I think the edit in this line is wrong.

Comment (samgermain) [reply js/gateio.js:653]: I was actually asking a question because I didn't know, so there's nothing to agree with me here
